### PR TITLE
fix: is online is only online if libp2p is online

### PR DESCRIPTION
**Pull request by pgte**
Also, when peer is discovered but not online yet (racing cond), delay dial until libp2p is online.

### CODE REVIEW

**Comment by pgte** [comment]
Should fix #874

File: src/core/components/libp2p.js [hunk 1]
```diff
@@ -25,10 +25,15 @@ module.exports = function libp2p (self) {
...
         self._libp2pNode.on('peer:discovery', (peerInfo) => {
-          if (self.isOnline()) {
+          const dial = () => {
             self._peerInfoBook.put(peerInfo)
             self._libp2pNode.dial(peerInfo, () => {})
           }
+          if (self.isOnline()) {
+            dial()
+          } else {
+            self._libp2pNode.once('online', dial)
```
Comment (diasdavid) [inline src/core/components/libp2p.js]: Does this event exist?

Comment (pgte) [inline src/core/components/libp2p.js]: Oops, I knew I had another PR to submit...
See https://github.com/libp2p/js-libp2p/pull/93

**Comment by pgte** [comment]
Depends on PRs:

- [ ] js-libp2p-floodsub - https://github.com/libp2p/js-libp2p-floodsub/pull/47
- [ ] libp2p-kad-dht - https://github.com/libp2p/js-libp2p-kad-dht/pull/12
- [ ] libp2p - https://github.com/libp2p/js-libp2p/pull/100

**Pull request merged**